### PR TITLE
Trench Warfare

### PR DESCRIPTION
**Pull request by Brottweiler**
- [x] Make kits for all classes
- [x] Make regions
- [x] Keep the KoTH?

### CODE REVIEW

File: Trench Warfare/map.xml [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            <block>smooth brick</block>
+            <block>iron fence</block>
+            <block>smooth stairs</block>
+            <block>wood button</block>
+        </deny>
+    </filter>
+</filters>
+<regions>
+    <!--Playing Field Regions-->
+    <union name="bases">
```
Comment (CrazyisCreeps) [inline Trench Warfare/map.xml]: In proto 1.4.0, the `name` attribute in regions is replaced with `id`. Looks like you have a mix of the two in the xml.

File: Trench Warfare/map.xml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            <item slot="2" amount="5">grilled pork</item>
+            <helmet material="diamond helmet"/>
+            <chestplate material="iron chestplate"/>
+            <leggings material="iron leggings"/>
+            <boots material="iron boots"/>
+        </kit>
+    </class>
+</classes>
+<filters>
+    <filter id="only-red">
```
Comment (CrazyisCreeps) [inline Trench Warfare/map.xml]: Again, another proto 1.4.0 behaviour, the way filters are done up changed between 1.3 and 1.4. Refer to something like https://github.com/MCResourcePile/stratus-maps/blob/master/maps/clay_soldiers/map.xml#L60 on how filters should look. The big thing is that individual filters aren't enclosed with `<filter>`

Comment (Brottweiler) [reply Trench Warfare/map.xml]: Thanks, I don't think I fixed it properly yet but I got rid of the `<filter>`.

Comment (CrazyisCreeps) [reply Trench Warfare/map.xml]: Yea looks good. The only issue I see left with filters is line 64; when listing multiple materials within a filter, surround them with `<any>`. So `<not><any><material 1><material 2></any></not>`

**Comment by Brottweiler** [comment]
Everything works except placing/breaking blocks in the field, and the TNT stuff.

Then gotta add the kits :smile:

File: Trench Warfare/map.xml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        <rectangle id="blue-home-side-right" min="-60,-11" max="-45,-17"/>
+    </apply>
+    <apply block="never">
+        <region id="blue-home"/>
+        <region id="red-home"/>
+    </apply>
+    <apply block-break="break-restrictions" block="no-tnt">
+        <complement>
+            <region id="playable"/>
+            <region id="red-home"/>
```
Comment (CrazyisCreeps) [inline Trench Warfare/map.xml]: What's the difference between `red-home` and `red-base`? It seems like you define the same region twice, or am I misreading? Same deal with blue.

File: Trench Warfare/map.xml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        </kit>
+    </class>
+</classes>
+<filters>
+    <team id="only-red">reds</team>
+    <team id="only-blue">blues</team>
+    <any id="only-tnt">
+        <material>tnt</material>
+    </any>
+    <not id="break-restrictions"> <!--Deny breaking stone brick and iron objects-->
```
Comment (CrazyisCreeps) [inline Trench Warfare/map.xml:212]: Looks all good! Only thing I might add is TNT to this list of block restrictions, that way preventing players from placing down TNT in the mid area of the map.

Unless of course the intent is to allow that to happen and I misunderstood.

File: Trench Warfare/map.xml [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+            <helmet material="diamond helmet" unbreakable="true"/>
+            <chestplate material="iron chestplate" unbreakable="true"/>
+            <leggings material="iron leggings" unbreakable="true"/>
+            <boots material="iron boots" unbreakable="true"/>
+        </kit>
+    </class>
+    <class name="Demoman" description="Specializes in high explosive materials." icon="sulphur">
+        <kit>
+            <item slot="0" unbreakable="true" material="stone sword"/>
+            <item projectile="gun" name="`aGrenade Launcher" material="stick"/>
```
Comment (Brottweiler) [inline Trench Warfare/map.xml]: Suggestion for item to use instead of `stick` is welcome.

File: Trench Warfare/map.xml [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+        <captured><cuboid min="-26,62,-7" max="-22,69,9"/></captured>
+    </control-point>
+</control-points>
+<projectiles>
+    <projectile
+        id="gun"
+        name="gun"
+        projectile="TNTPrimed"
+        velocity="0.5"
+        damage="20"
```
Comment (Brottweiler) [inline Trench Warfare/map.xml]: `0.5` velocity seems good, almost just "throwing" out a TNT, and grenade launcher don't shoot very far I think

Not sure how much damage it should do though. I think it should maybe do less than what a normal TNT do.

**Comment by Brottweiler** [comment]
Merged in https://github.com/OvercastCommunity/maps/commit/a5d393d063a7e24154c1516e7f9b9c9bf7e15408 and I realised I completely screwed up the commit message :woman_facepalming: oh well it happens

**Pull request closed without merging**